### PR TITLE
#240 Build IPA after signing app in linking phase

### PR DESCRIPTION
**Pull request by jperedadnr**

### CODE REVIEW

**Comment by eugener** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 3
           

Complexity increasing per file
==============================
- src/main/java/com/gluonhq/substrate/target/IosTargetConfiguration.java  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/gluonhq/substrate/pullRequest?prid=4769192&bid=15895310)

File: src/main/java/com/gluonhq/substrate/util/ios/CodeSigning.java [hunk 1]
```diff
@@ -93,6 +92,9 @@ public CodeSigning(ProcessPaths paths, InternalProjectConfiguration projectConfi
 
         appPath = paths.getAppPath().resolve(projectConfiguration.getAppName() + ".app");
         tmpPath = paths.getTmpPath();
+
+        providedIdentityName = projectConfiguration.getIosSigningConfiguration().getProvidedSigningIdentity();
```
Comment (eugener) [inline src/main/java/com/gluonhq/substrate/util/ios/CodeSigning.java]: Codacy Issue found: [Possible unsafe assignment to a non-final static field in a constructor.](https://app.codacy.com/manual/gluonhq/substrate/pullRequest?prid=4769192)

File: src/main/java/com/gluonhq/substrate/target/IosTargetConfiguration.java [hunk 2]
```diff
@@ -155,6 +155,9 @@ public boolean link() throws IOException, InterruptedException {
                 if (!codeSigning.signApp()) {
                     throw new RuntimeException("Error signing the app");
                 }
+                if (!buildIPA()) {
+                    throw new RuntimeException("Error building the IPA");
```
Comment (eugener) [inline src/main/java/com/gluonhq/substrate/target/IosTargetConfiguration.java]: Codacy Issue found: [Avoid throwing raw exception types.](https://app.codacy.com/manual/gluonhq/substrate/pullRequest?prid=4769192)

File: src/main/java/com/gluonhq/substrate/util/ios/CodeSigning.java [hunk 3]
```diff
@@ -93,6 +92,9 @@ public CodeSigning(ProcessPaths paths, InternalProjectConfiguration projectConfi
...
+        providedMobileProvision = projectConfiguration.getIosSigningConfiguration().getProvidedProvisioningProfile();
```
Comment (eugener) [inline src/main/java/com/gluonhq/substrate/util/ios/CodeSigning.java]: Codacy Issue found: [Possible unsafe assignment to a non-final static field in a constructor.](https://app.codacy.com/manual/gluonhq/substrate/pullRequest?prid=4769192)

File: src/main/java/com/gluonhq/substrate/target/IosTargetConfiguration.java [hunk 4]
```diff
@@ -155,6 +155,9 @@ public boolean link() throws IOException, InterruptedException {
                 if (!codeSigning.signApp()) {
                     throw new RuntimeException("Error signing the app");
                 }
+                if (!buildIPA()) {
```
Comment (johanvos) [inline src/main/java/com/gluonhq/substrate/target/IosTargetConfiguration.java]: shouldn't be done at linking, but at packaging task

**Comment by eugener** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/gluonhq/substrate/pullRequest?prid=4769192&bid=15895310)

File: src/main/java/com/gluonhq/substrate/util/ios/CodeSigning.java (hunk 1 above)
Comment (eugener) [inline src/main/java/com/gluonhq/substrate/util/ios/CodeSigning.java]: Codacy Issue found: [Possible unsafe assignment to a non-final static field in a constructor.](https://app.codacy.com/manual/gluonhq/substrate/pullRequest?prid=4769192)

File: src/main/java/com/gluonhq/substrate/util/ios/CodeSigning.java (hunk 3 above)
Comment (eugener) [inline src/main/java/com/gluonhq/substrate/util/ios/CodeSigning.java]: Codacy Issue found: [Possible unsafe assignment to a non-final static field in a constructor.](https://app.codacy.com/manual/gluonhq/substrate/pullRequest?prid=4769192)

**Comment by eugener** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/gluonhq/substrate/pullRequest?prid=4769192&bid=15895310)

**Pull request merged**